LibreNMS/Authentication/RadiusAuthorizer.php

```diff
@@ -33,7 +33,27 @@ public function authenticate($credentials)
 
         $password = $credentials['password'] ?? null;
         if ($this->radius->accessRequest($credentials['username'], $password) === true) {
-            $this->addUser($credentials['username'], $password, Config::get('radius.default_level', 1));
+            if ($this->userExists($credentials['username'])) {
+
+                //attribute 11 is "Filter-Id"
+                //Always set password change to 0 - password resides in AAA, not LibreNMS
+                //If attribute 11 is sent in reply after accept - update user
+                //If user exists - update, not add.
+                //If new user - add user with attribute value if present, or use default from config.
+                if ($this->radius->getAttribute(11)) {
+                    $this->updateUser($this->getUserid($credentials['username']), $credentials['username'], intval($this->radius->getAttribute(11)), 0, '');
+                } else {
+                    $this->updateUser($this->getUserid($credentials['username']), $credentials['username'], Config::get('radius.default_level', 1), 0, '');
+                }
+            }
+
+            if (! $this->userExists($credentials['username'])) {
```

Comment:
Wouldn't this be logically the same as plain Else?

Reply:
Maybe, but one would need to test for 4 scenarios - and it made more sense to me to do 2 if/else blocks, than put it all together.

You will get one of these four cases:

New user with attribute
New user with no attribute

Existing user with attribute
Existing user without attribute.

As long as librenms saves the user as a regular user, i believe this checks are needed? If they only were saved temporary and deleted at logout - there would only be necessary with one check.